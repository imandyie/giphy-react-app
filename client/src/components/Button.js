import React, { PureComponent } from 'react';
import { PropTypes } from 'prop-types';
import '../styles/components/Button.scss';
import colors from '../styles/colors';

export default class Button extends PureComponent {
  render() {
    const {
      text,
      color,
      borderColor,
      type,
      className,
      isFetching,
      onClick,
    } = this.props;
    const buttonType = type || 'primary';
    const customClassName = className || '';
    const fetchingClass = isFetching ? 'animated-button-placeholder' : '';

    return (
      <button
        className={`button ${buttonType}-button ${customClassName} ${fetchingClass}`}
        style={{borderColor, color}}
        onClick={onClick}
      >
        {text}
      </button>
    );
  }
}

Button.propTypes = {
  text: PropTypes.string.isRequired,
  type: PropTypes.string,
  borderColor: PropTypes.string,
  color: PropTypes.string,
  className: PropTypes.string,
  isFetching: PropTypes.bool,
  onClick: PropTypes.func.isRequired
};
