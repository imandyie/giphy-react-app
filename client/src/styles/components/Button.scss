@import '../colors.scss';

.button {
  background-color: transparent;
  padding: 10px 20px;
  outline: 0;
  cursor: pointer;
  font-size: 0.8rem;

  &.primary-button {
    border: 1px solid;
    border-radius: 20px;
  }

  &:hover {
    opacity: 0.6;
  }
}

@keyframes buttonPlaceholder {
  0% {
    background-position: -1200px 0
  }
  100% {
    background-position: 1200px 0
  }
}

.animated-button-placeholder {
  animation-duration: 2.1s;
  animation-fill-mode: forwards;
  animation-iteration-count: infinite;
  animation-name: buttonPlaceholder;
  animation-timing-function: linear;
  background: $white;
  background: linear-gradient(to right, $darkerGray 8%, $darkGray 18%, $darkerGray 33%);
  background-size: 800px 104px;
  border-color: $darkerGray !important;
  color: transparent !important;
  margin-top: 15px;
  margin-bottom: 35px;
}