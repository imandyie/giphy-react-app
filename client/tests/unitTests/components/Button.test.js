import React from 'react';
import Button from '../../../src/components/Button';
import renderer from 'react-test-renderer';

test('Button renders without errors', () => {
  const component = renderer.create(
    <Button
      text="Sample Button"
      onClick={() => { alert('hello') }}
      borderColor="red"
      color="blue"
    />,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});