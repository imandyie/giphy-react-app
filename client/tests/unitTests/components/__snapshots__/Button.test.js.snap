// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders without errors 1`] = `
<button
  className="button primary-button  "
  onClick={[Function]}
  style={
    Object {
      "borderColor": "red",
      "color": "blue",
    }
  }
>
  Sample Button
</button>
`;
